backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java

```diff
@@ -0,0 +1,84 @@
+package com.canvas.service.helperServices;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import org.codehaus.plexus.util.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Filter class for header processing.
+ */
+@Component
+public class HeaderFilter implements Filter {
+
+    /**
+     * Constructor for dependency injection of AESCryptoService.
+     * @param aesCryptoService instance of AESCryptoService
+     */
+    @Autowired
+    private final AESCryptoService aesCryptoService;
+
+    @Autowired
+    public HeaderFilter(AESCryptoService aesCryptoService) {
```

Comment:
For testing, we should have a no arg constructor as well with static variables we can use. Please add one before this constructor.

Reply:
Same as [this|https://github.com/Canvas-Code-Capstone/Canvas-Code/pull/57/files/9bcdee2284a82ce7969f0d6edeb5515e45b6dd5c#r1113552167] comment. Will address in a future PR.